tao_compiler/mlir/disc/disc_compiler.cc

```diff
@@ -277,6 +265,18 @@ LogicalResult LowerHLOToLLVM(ModuleOp m, const DISCLoweringOptions& options) {
   pm.addNestedPass<FuncOp>(
       disc_ral::createDiscElementTypeConverterPass(enable_fp16));
 
+  pm.addNestedPass<FuncOp>(disc_ral::createDiscConvRewriter());
```

Comment:
Is there any special reason for which you move the amp pass forwad?

We convert mhlo::conv to mhlo::DynamicConv in this pass as well. So maybe we should re-consider the order between this pass and this `AMP` pass.

Reply:
ConvRewriter pass only transforms fp16 conv to NHWC for gpu, so the conversion from fp32 to fp16 should happen before this pass.

Reply:
nit: maybe we could consider split the `ConvRewriter` pass into two passes, one for mhlo::conv to mhlo::dynamic conv   and one for layout optimization?

Reply:
Is this still nesscessary if mhlo::ConvOp is also covered in AMP pass?

Reply:
I think it's ok as well. In general, it would be better if we could handle mhlo::conv and mhlo::dynamic_conv in one pass and then the other passes do not need to worry about the difference between these two ops.